lib/sidekiq/form526_backup_submission_process/processor.rb

```diff
@@ -0,0 +1,298 @@
+      attr_accessor :docs
+
+      FORM_526 = 'form526'
+      FORM_526_DOC_TYPE = '21-526EZ'
+      FORM_526_UPLOADS = 'form526_uploads'
+      FORM_526_UPLOADS_DOC_TYPE = 'evidence'
+      FORM_4142 = 'form4142'
+      FORM_4142_DOC_TYPE = '21-4142'
+      FORM_0781 = 'form0781'
+      FORM_8940 = 'form8940'
+      FLASHES = 'flashes'
+      BIRLS_KEY = 'va_eauth_birlsfilenumber'
+      TMP_FILE_PREFIX = 'form526.backup.'
+      EVIDENCE_LOOKUP = {}.freeze
+      BKUP_SETTINGS = Settings.key?(:form526_backup) ? Settings.form526_backup : OpenStruct.new
+
+      SUB_METHOD = (BKUP_SETTINGS.submission_method || 'single').to_sym
+      CONSUMER_NAME = 'vets_api_backup_submission'
+
+      # Takes a submission id, assembles all needed docs from its payload, then sends it to central mail via
+      # lighthouse benefits intake API - https://developer.va.gov/explore/benefits/docs/benefits?version=current
+      def initialize(submission_id, docs = [])
+        @submission = Form526Submission.find(submission_id)
+        @docs = docs
+        @lighthouse_service = Form526BackupSubmission::Service.new
+        # We need an initial location/uuid as other ancillary docs want a reference id to it
+        # (eventhough I dont think they actually use it for anything because we are just using them to
+        # generate the pdf and not the sending portion of those classes... but it needs something there to not error)
+        @initial_upload = get_upload_info
+        uuid_and_location = upload_location_to_location_and_uuid(initial_upload)
+        @initial_upload_uuid = uuid_and_location[:uuid]
+        @initial_upload_location = uuid_and_location[:location]
+        determine_zip
+      end
+
+      def process!
+        # Generates or makes calls to get, all PDFs, adds all to self.docs obj
+        gather_docs!
+        # Iterate over self.docs obj and add required metadata to objs directly
+        add_meta_data_to_docs!
+        # Take assemebled self.docs and aggregate and send how needed
+        send_to_central_mail_through_lighthouse_claims_intake_api!
+      end
+
+      private
+
+      # Transforms the lighthouse response to the only info we actually need from it
+      def upload_location_to_location_and_uuid(upload_return)
+        {
+          uuid: upload_return.dig('data', 'id'),
+          location: upload_return.dig('data', 'attributes', 'location')
+        }
+      end
+
+      # Instantiates a new location and uuid via lighthouse
+      def get_upload_info
+        lighthouse_service.get_upload_location.body
+      end
+
+      # Combines instantiating a new location/uuid and returning the important bits
+      def get_upload_location_and_uuid
+        upload_info = get_upload_info
+        upload_location_to_location_and_uuid(upload_info)
+      end
+
+      def received_date
+        date = SavedClaim::DisabilityCompensation.find(submission.saved_claim_id).created_at
+        date = date.in_time_zone('Central Time (US & Canada)')
```

Comment:
It might be worth putting a comment here to explain why it needs to be in central time (I'm guessing this is something to do with EVSS servers being in central time?)

Reply:
I am not actually certain WHY this is the case. This was taken from other EVSS submissions, but I will see what I can dig up.

Reply:
Yes, those evss servers run in central time is my understanding, someone told me they are in Texas somewhere.....